### ISSUE TITLE
Rest Asset Search has no output with empty request parameter: maven.classifier

### ISSUE THREAD

**Issue by svenfischerBR** [sonatype/nexus-public#188]
* What problem are you trying to solve?
we have upgraded nexus from 3.52.0 to 3.57.0
after upgrade the following Request:
http://localhost/service/rest/v1/search/assets/download?group=com.example.test&name=example-artefact&version=1.0.1&maven.extension=jar&sort=version&maven.classifier

gets an empty result:
```
{

      "items": [ ],
      "continuationToken": null

}
```

without the request parameter: maven.classifier 

we get 2 items: 
```
...
"maven2": {

            "extension": "jar",
            "groupId": "com.example.test",
            "classifier": "sources",
            "artifactId": "example-artefact",
            "version": "1.0.1"
}
....
"maven2": {

            "extension": "jar",
            "groupId": "com.example.test"",
            "artifactId": "example-artefact",
            "version": "1.0.1"
}
```
 

If we set the request parameter: "maven.classifier=sources" - as expected, we get the 
sources item. And if the request is done on an artifact, that has only one artifact (no sources or docs) the
result is also as expected.

* Do you have a workaround you are using at present? no
* What feature or behavior is this required for?
curl Artifact File, especially from snapshot repositories, to curl the newest artifact

* How could we solve this issue? not knowing
* Tell us about your Nexus Repository deployment: what version, operating system, and database are you using?
- CentOS Linux release 7.9.2009
- Buildin DB
- Version: 3.57.0

* Anything else? we have created the request as documented here: https://help.sonatype.com/repomanager3/integrations/rest-and-integration-api/search-api#SearchAPI-SearchandDownloadAsset 

Hope you can help me with this problem.

Thanks

Sven

**Comment by tim-tk** [comment]
Same problem here. Not the first time our API calls got broken with an update...

**Comment by nblair** [comment]
Hi @svenfischerBR thanks for reporting this issue. We can confirm the reported behavior and have good news to report that a fix has been completed and is expected to be included in release 3.59.0 which is expected on August 8, 2023.

**Comment by nblair** [comment]
Apologies for the delay, but 3.59.0 was released this past Tuesday, August 15 and includes the fix (see the [release notes](https://help.sonatype.com/repomanager3/product-information/release-notes/2023-release-notes/sonatype-nexus-repository-3.59.0-release-notes)). Let us know if it helped!
